common/txmgr/types/fee_estimator.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import "context"
+
+// Opt is an option for a gas estimator
+type Opt int
+
+const (
+	// OptForceRefetch forces the estimator to bust a cache if necessary
+	OptForceRefetch Opt = iota
+)
+
+// PriorAttempt provides a generic interface for reading tx data to be used in the fee esimators
+type PriorAttempt[FEE any, HASH any] interface {
```

Comment:
Btw, I don't think there is any reason for the common TxMgr components to use PriorAttempt interface.
It already doesn't use it, except to pass it to Estimator.BumpGas().
I believe we could always just pass in previous Attempts in the form of a generic TxAttempt. And the chain specific code should know what to extract from it, to bump gas correctly.

Not for this PR, but for later, let us aim to completely do away with this interface, and just reuse TxAttempt.

If you want, create a separate subtask or ticket for this, for later.

Reply:
👍🏼 I will take a look at this part when I'm working with the TxBuilder